pandas/core/generic.py

```diff
@@ -1906,23 +1906,54 @@ def to_pickle(self, path, compression='infer',
         Parameters
         ----------
         path : string
-            File path
+            File path where the pickled object will be stored.
         compression : {'infer', 'gzip', 'bz2', 'xz', None}, default 'infer'
-            a string representing the compression to use in the output file
+            A string representing the compression to use in the output file. By
+            default, infers from the specified path.
 
             .. versionadded:: 0.20.0
         protocol : int
             Int which indicates which protocol should be used by the pickler,
             default HIGHEST_PROTOCOL (see [1], paragraph 12.1.2). The possible
             values for this parameter depend on the version of Python. For
             Python 2.x, possible values are 0, 1, 2. For Python>=3.0, 3 is a
-            valid value. For Python >= 3.4, 4 is a valid value.A negative value
-            for the protocol parameter is equivalent to setting its value to
-            HIGHEST_PROTOCOL.
+            valid value. For Python >= 3.4, 4 is a valid value. A negative
+            value for the protocol parameter is equivalent to setting its value
+            to HIGHEST_PROTOCOL.
 
             .. [1] https://docs.python.org/3/library/pickle.html
```

Comment:
I think is more standard to have this in a `References` section.

Reply:
@jorisvandenbossche your opinion?

Reply:
I previously said it was fine, because we don't use `References` sections in many cases (most of the time we use inline links), another thing we can discuss in further improving the guidelines.